runtime/runtimecontextmanager.go

```diff
@@ -118,12 +128,27 @@ func (m *runtimeContextManager) PushContext(ctx RuntimeContextDef) {
 	m.status = StatusLive
 	m.messageHandler = ctx.MessageHandler
 	m.parent = &parent
+	if ctx.GCPolicy == IsolateGCPolicy || ctx.HardLimits.Millis > 0 || ctx.HardLimits.Cpu > 0 || ctx.HardLimits.Memory > 0 {
+		m.weakRefPool = weakref.NewDefaultPool()
```

Comment:
What happens if you already have a pool, do you close that pool before pushing a context? See initRoot, which sets up a pool.

Reply:
If you already have a pool, then you don't close it but it becomes "inactive" until the new context is popped.